File: Ahorn/entities/TowerGargoyle.jl
module TowerGargoyle

using ..Ahorn, Maple

@mapdef Entity "MaggyHelper/TowerGargoyle" TowerGargoyle(x::Integer, y::Integer, detectionRange::Integer=150, glideSpeed::Integer=100, health::Integer=2, swoopSpeed::Integer=250)

const placements = Ahorn.PlacementDict(
    "normal" => Ahorn.EntityPlacement(TowerGargoyle)
)

function Ahorn.selection(entity::TowerGargoyle)
    return Ahorn.Rectangle(entity.x - 8, entity.y - 8, 16, 16)
end

function Ahorn.render(ctx::Ahorn.Cairo.CairoContext, entity::TowerGargoyle, room::Maple.Room)
    Ahorn.drawRectangle(ctx, Ahorn.Rectangle(entity.x - 8, entity.y - 8, 16, 16), (0.5, 0.5, 0.5, 0.5), (0.0, 0.0, 0.0, 1.0))
end

end

File: Loenn/entities/towerGargoyle.lua
local utils = require("utils")
local drawableSprite = require("structs.drawable_sprite")

local towerGargoyle = {}

towerGargoyle.name = "MaggyHelper/TowerGargoyle"
towerGargoyle.depth = -100
towerGargoyle.fieldInformation = {
    health = {
        minimumValue = 1
    },
    detectionRange = {
        minimumValue = 0
    },
    swoopSpeed = {
        minimumValue = 0
    },
    glideSpeed = {
        minimumValue = 0
    }
}
towerGargoyle.placements = {
    {
        name = "normal",
        data = {
            health = 2,
            detectionRange = 150,
            swoopSpeed = 250,
            glideSpeed = 100
        }
    }
}

function towerGargoyle.sprite(room, entity)
    local sprite = drawableSprite.fromTexture("characters/tower_gargoyle/dormant", entity)
    sprite:setPosition(entity.x, entity.y)
    sprite:setJustification(0.5, 1.0)
    return sprite
end

function towerGargoyle.selection(room, entity)
    return utils.rectangle(entity.x - 14, entity.y - 24, 28, 24)
end

return towerGargoyle
